src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -465,6 +469,11 @@ public MethodSymbol OverriddenMethod
             }
         }
 
+        /// <summary>
+        /// If we know a specific overridden class method, that method.
+        /// </summary>
+        internal virtual MethodSymbol ExplicitlyOverriddenClassMethod => null;
```

Comment:
>internal virtual MethodSymbol ExplicitlyOverriddenClassMethod => null; [](start = 8, length = 70)

We have learned over the years that introducing new virtual methods at the top of the symbol hierarchy is error prone. It is very easy to fail to provide a required override for some derived type when the method is introduced. And it is even easier to do that when some derived type is added later. Our current approach is to add abstract methods and provide implementation in all leaf classes. This way we can review that the method is properly implemented for all flavors and can test all relevant scenarios by using code coverage.

Reply:
This method is removed later in this PR.

---
In reply to: [424636776](https://github.com/dotnet/roslyn/pull/44025#discussion_r424636776) [](ancestors = 424636776)